File: config/config-sbc2ha/src/main/java/com/dfi/sbc2ha/config/sbc2ha/definition/sensor/analog/ResistanceConfig.java
package com.dfi.sbc2ha.config.sbc2ha.definition.sensor.analog;

import com.dfi.sbc2ha.config.sbc2ha.definition.enums.PlatformType;
import com.dfi.sbc2ha.config.sbc2ha.definition.enums.ResistanceDirectionType;
import com.dfi.sbc2ha.config.sbc2ha.definition.sensor.SingleSourceConfig;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ResistanceConfig extends SingleSourceConfig {

    /**
     * The type of circuit, one of DOWNSTREAM or UPSTREAM.
     * required
     */
    ResistanceDirectionType direction;

    /**
     * The value of the resistor with a constant value.
     * required
     */
    String resistor;

    /**
     * The reference voltage. Defaults to 1.8V.
     * optional
     */
    @JsonProperty("reference_voltage")
    String referenceVoltage;

    public ResistanceConfig(String sensor) {
        super(sensor);
        platform = PlatformType.RESISTANCE;
    }

    public ResistanceConfig() {
        super();
        platform = PlatformType.RESISTANCE;
    }
}

File: config/editor/src/app/definition/sensor/analog/ResistanceConfig.ts
import {PlatformType} from "../../enums/PlatformType";
import {ResistanceDirectionType} from "../../enums/ResistanceDirectionType";
import {SingleSourceConfig} from "../SingleSourceConfig";
import {Expose} from "class-transformer";


export class ResistanceConfig extends SingleSourceConfig {
    direction: ResistanceDirectionType;
    resistor: string;
    @Expose({name: "reference_voltage"})
    referenceVoltage: string;

    constructor(name: string, sensor: string, direction: ResistanceDirectionType, resistor: string, referenceVoltage: string) {
        super(PlatformType.RESISTANCE, name, sensor);
        this.direction = direction;
        this.resistor = resistor;
        this.referenceVoltage = referenceVoltage;
    }
}
